File: BlockTutorial/BlockTutorial/ViewController.m
//
//  ViewController.m
//  BlockTutorial
//
//  Created by w91379137 on 2016/6/27.
//  Copyright © 2016年 w91379137. All rights reserved.
//

#import "ViewController.h"
#import "HTTPBinManager.h"

@interface ViewController ()
<HTTPBinManagerDelegate>
{
    IBOutlet UIImageView *imageView;
    IBOutlet UIProgressView *progressViewr;
    HTTPBinManager *manager;
}

@end

@implementation ViewController

#pragma mark - Init
- (void)viewDidLoad
{
    [super viewDidLoad];
}

#pragma mark - IBAction
- (IBAction)startAction:(id)sender
{
    manager = [HTTPBinManager sharedInstance];
    manager.delegate = self;
    [manager executeOperation];
}

#pragma mark - HTTPBinManagerDelegate
- (void)operationManagerNotice:(HTTPBinManager *)binManager
{
    imageView.image = binManager.image;
    progressViewr.progress = binManager.progress;
}

@end

File: GCDProgramming23/GCDProgramming23/ViewController.swift
//
//  ViewController.swift
//  GCDProgramming23
//
//  Created by w91379137 on 2016/6/28.
//  Copyright © 2016年 w91379137. All rights reserved.
//

import UIKit

class ViewController: UIViewController, HTTPBinManagerDelegate {
    
    //MARK: - Property
    
    @IBOutlet var imageView : UIImageView!
    let manager = HTTPBinManager()
    
    //MARK: - Init
    override func viewDidLoad() {
        super.viewDidLoad()
        
        manager.delegate = self
        manager.executeOperation()
    }
    
    //MARK: - HTTPBinManagerDelegate
    func operationManagerNotice(binManager: HTTPBinManager!)
    {
        if binManager.cancelled {
            print("取消 取消前進度\(binManager.progress)")
        }
        else {
            print("繼續進行 目前進度\(binManager.progress)")
//            if binManager.progress > 0.2 && !binManager.cancelled {
//                binManager.cancelOperation()
//            }
        }
    }
}

